### ISSUE TITLE
Generic.reseed doesn't seem to work on internal providers

### ISSUE THREAD

**Issue by ronniehicks** [lk-geimfari/mimesis#1115]
# Bug report

## What's wrong

When using the generic provider it seems the reseed function doesn't work correctly.

```python
from mimesis import Generic
g = Generic(seed=1234)
g.address.address()

g.reseed(1234)
g.address.address()
```

Notice that the address value printed isn't the same after reseed.

## How is that should be

Address printed after reseed should match address printed before.

## System information

MacOS 10.15.7
Python 3.7.10
Mimesis 4.1.3 (I know this isn't latest, but my current use case requires an older version of python)

**Comment by lk-geimfari** [comment]
Well, seems like a bug. I can offer temporary workaround:

```python
g = Generic(seed=1234)
g.address.address()
# Output: 903 Constanso Alley
g.address.reseed(1234)
g.address.address()
# Output: 903 Constanso Alley
```

**Comment by lk-geimfari** [comment]
Or, something like this:

```python
class SeedGeneric(Generic):
    def reseed(self, seed: Optional[Seed] = None) -> None:
        for attr_name in self.__dir__():
            try:
                provider = getattr(self, attr_name)
                provider.reseed(seed=seed)
            except AttributeError:
                continue
                
g = SeedGeneric(seed=1234)
g.address.address()
# Output: 903 Constanso Alley
g.reseed(1234)
g.address.address()
# Output: 903 Constanso Alley
```

**Comment by lk-geimfari** [comment]
I've fixed it in #94eee4e, so I've closed this issue.
